squid/templates/deployment.yaml

```diff
@@ -67,12 +67,52 @@ spec:
             - name: http
               containerPort: 3128
               protocol: TCP
+            {{- if .Values.squidExporter.enabled }}
+            - name: metrics
+              containerPort: {{ .Values.squidExporter.port }}
+              protocol: TCP
+            {{- end }}
+          {{- if .Values.squidExporter.enabled }}
+          command: ["/bin/sh"]
+          args:
+            - -c
+            - |
+              # Start squid-exporter in background
+              /usr/local/bin/squid-exporter \
+                -squid-hostname localhost \
+                -squid-port {{ .Values.service.port }} \
+                -listen :{{ .Values.squidExporter.port }} \
+                -metrics-path {{ .Values.squidExporter.metricsPath }} \
+                {{- if .Values.squidExporter.squidLogin }}
+                -squid-login "{{ .Values.squidExporter.squidLogin }}" \
+                {{- end }}
+                {{- if .Values.squidExporter.squidPassword }}
+                -squid-password "{{ .Values.squidExporter.squidPassword }}" \
+                {{- end }}
+                {{- range $key, $value := .Values.squidExporter.customLabels }}
+                -label "{{ $key }}={{ $value }}" \
+                {{- end }}
+                &
+              
+              # Start squid and pipe access logs to squid-exporter via stdin
+              exec /usr/sbin/container-entrypoint.sh
```

Comment:
This is an anti-pattern because one running container should only ever map to a one process. We can continue to run the squid-exporter in a separate container while using the same image.